coderd/database/queries/insights.sql

```diff
@@ -329,6 +329,267 @@ SELECT
 FROM unflattened_usage_by_interval
 GROUP BY from_, to_;
 
+-- name: GetTemplateUsageStats :many
```

Comment:
These queries 'look OK' to me, but I can't honestly say that I understand them at a fundamental level. Debugging issues with these queries is likely to require significant effort. I wonder if we could take an existing database dump, sanitize it, and use that in a golden-style test?

Reply:
We don't use a dump per-se, but we already do this with existing tests by defining a few templates, workspaces, users, and apps, inserting agent and app stats for predefined times when they "used" the workspace, and finally doing golden-file verification against the JSON produced by the API endpoints.

Reply:
Fair enough. 👍